projects/output-and-error-handling/client/main_test.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+)
+
+// This is the beginning of the test implementation
+// It is not comprehensive by any means
+
+func TestParseResponseBody(t *testing.T) {
```

Comment:
These tests are all very reasonable as far as using the standard library goes, but they all feel a little repetitive and verbose in the frequent pattern of:

```go
if want != got {
    t.Errorf("got %q | want %q", got, want")
}
```

I'd encourage you to take a look at [`require`, a library for test assertions](https://pkg.go.dev/github.com/stretchr/testify/require) which I'm a big fan of - it allows you to just write:

```go
require.Equal(t, want, got)
```

which is much less repetitive.

The common functions I tend to use from it are: `require.Equal`, `require.NoError`, `require.ErrorContains`, and `require.Contains`.